ftplugin/markdown.vim

```diff
@@ -48,20 +48,20 @@
 " For each level, contains the regexp that matches at that level only.
 "
 let s:levelRegexpDict = {
-    \ 1: '\v^(#[^#]@=|.+\n\=+$)',
-    \ 2: '\v^(##[^#]@=|.+\n-+$)',
-    \ 3: '\v^###[^#]@=',
-    \ 4: '\v^####[^#]@=',
-    \ 5: '\v^#####[^#]@=',
-    \ 6: '\v^######[^#]@='
+    \ 1: '\v^(#[^#]@=\s+|.+\n\=+$)',
```

Comment:
* CommonMark ATX heading can be empty. https://spec.commonmark.org/0.30/#example-79.
* I think it's not necessary to include the whitespace in the match. 
```suggestion
    \ 1: '\v^(#(\_s+)@=|.+\n\=+$)',
```